jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java

```diff
@@ -180,26 +183,30 @@ public Authorization authenticatePush(@Nullable Credential credential)
   }
 
   @VisibleForTesting
-  String getServiceScopeRequestParameters(String scope) {
-    return "service="
-        + service
-        + "&scope=repository:"
-        + registryEndpointRequestProperties.getImageName()
-        + ":"
-        + scope;
+  String getServiceScopeRequestParameters(Map<String, String> repositoryScopes) {
+    StringBuilder parameters = new StringBuilder("service=").append(service);
+    for (Entry<String, String> pair : repositoryScopes.entrySet()) {
```

Comment:
Do we expect this to only return what is necessary? Is it possible this generates a string with some unused scopes?

Reply:
In theory it could contain all sorts of scopes.  But in practice the `repositoryScopes` will contain at most two:

  1. the target repository &rarr; `push,pull`
  2. *iff on the same registry*, then the source repository &rarr; `pull`